src/Components/ProjectsComponent/ScheduleComponents/ScheduleViewComponent.jsx

```diff
@@ -4,76 +4,79 @@ import PropTypes from "prop-types";
 import { ViewComponent } from "Components/CustomComponents";
 import { AddButton } from "Components/forms/Buttons";
 import NoResultsFound from "Components/NoResultsFound";
-import TableComponent from "Components/CustomComponents/TableComponent/index";
+import TableComponent from "Components/CustomComponents/TableComponent";
 
-const ScheduleViewComponent = (props) => {
-  const { data, isLoading, onNavigate } = props;
+import ScheduleTableRow from "./components/ScheduleTableRow";
 
-  const AddNew = () => {
-    onNavigate("new");
-  };
+const tableHeadings = [
+  "#",
+  "Name",
+  "Type",
+  "Environment",
+  "Frequency",
+  "Created on",
+  "Status",
+  "Scheduled By",
+];
+
+const ScheduleViewComponent = ({ data, navigateToScheduleForm }) => {
+  const viewComponentRight = (
```

Comment:
can we move this out side this functional component

Reply:
It is dependent on props as we are passing `navigateToScheduleForm` props in the AddButton

Reply:
In this case if data change viewComponentRight will get new address so it will be rerendered again and again.

Reply:
so in that case we can use the useMemo hook, right?

Reply:
Ideally we should kept it outside

Reply:
Let's consider we kept it outside and pass to viewComponent
```suggestion
  const ViewComponentRight = ({ navigateToScheduleForm }) => (
    <AddButton label="Schedule New" handleClick={navigateToScheduleForm} />
  );
  const ScheduleViewComponent = ({ data, navigateToScheduleForm }) => {
    return (
      <ViewComponent
        title="Schedule Testcases"
        hideBtns
        rightChildrens={
          <ViewComponentRight navigateToScheduleForm={navigateToScheduleForm} />
        }
      >
```

But I think this doesn't change anything it will also get a new address when data changes.

Reply:
https://www.youtube.com/watch?v=QuLfCUh-iwI
Watch mistake no 2

Reply:
code optimized by using `useMemo` and `useCallback`.